src/main/java/com/internetitem/logback/elasticsearch/util/ContextMapWriter.java

```diff
@@ -0,0 +1,62 @@
+package com.internetitem.logback.elasticsearch.util;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import com.fasterxml.jackson.core.JsonGenerator;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+public class ContextMapWriter {
+
+    public void writeContextMap(JsonGenerator gen, ILoggingEvent event) throws IOException {
+        Object[] arguments = event.getArgumentArray();
+        if (arguments == null || arguments.length == 0) return;
+        Object lastElement = arguments[arguments.length - 1];
+        if (lastElement instanceof Map) {
+            Map<String, Object> indexes = traverseObject(new HashMap<String, Object>(), "context", lastElement);
```

Comment:
I would suggest to call `traverseMap` directly here. It took me a bit to figure out why `traverseObject` is performing `instanceof` checks again, when `lastElement` is already guaranteed to be a map here.